lms/djangoapps/verify_student/views.py

```diff
@@ -269,15 +269,27 @@ def get(
         )
         if redirect_url:
             return redirect(redirect_url)
+        course_mode = None
+        expired_course_mode, unexpired_paid_course_mode = self._get_expired_verified_and_all_paid_modes_for_course(
+            course_key)
+
+        if unexpired_paid_course_mode:
+            course_mode = unexpired_paid_course_mode[0]
```

Comment:
Since you are only using the first unexpired paid mode, maybe you could move this logic into the helper function, so that the helper function returns just the one paid mode it found (or None):
```
def _get_expired_verified_and_paid_mode():
    ...
   return (expired_verified, paid_mode)
```